index.bs

```diff
@@ -1877,6 +1877,40 @@ If the bytes in the buffer have a natural intepretation
 as one of the other TypedArray types, provide that instead.
 For example, if the bytes represent Float32 values, use a {{Float32Array}}.
 
+<h3 id="allow-microtask-switches">A Promise represents a value, not a callback</h3>
+
+Avoid requiring that something needs to happen synchronously within
+a promise resolution or rejection callback.
+
+Authors tend to expect that any `async` function can be wrapped
+in another that appends synchronous code at the end e.g. to inspect state
+using try/catch. But the `await` queues a microtask:
+<div class="example">
+```js
+async function fooWrapper() {
+  const foo = await platform.foo();
+  console.log("success");
+  return foo;
+}
+
+(async () => {
+  const foo = await fooWrapper();
+  foo.bar(); // on the same task, but not the same microtask that logged "success"
+})();
+```
+</div>
+
+To support this pattern, a specification that wishes to restrict `bar()` to
+only succeed if called immediately following a promise resolving can
+minimally limit it to the same task, but not the same microtask.
```

Comment:
Ideally, we should discourage any such limitations first.  Then we talk about why a platform API might have cause to add a restriction like this.

I would say instead something like:

> In general, the return value from a function should be usable at any time.
>
> There could be cases where APIs might need to restrict the scope over which a value can be used.
> In the above example, perhaps the `platform.foo()` function establishes state that has real-time dependencies
> such that calling `bar()` is not viable at any future time.
>
> Never limit the validity of returned values to a microtask, as this prevents most forms of code composition.
> If possible, set a short timeout interval.
> Failing that, returned values can be limited to the same task, but not a single microtask.

Reply:
Totally agree. Just a bit concerned about the phrase "returned value", given the subtleties involved (technically the promise is the returned value).

Maybe s/returned value/resolved value/ ?

Reply:
Or maybe "settled result" since this is less about the value and more about the timing of the callback. E.g. some API devising some synchronous contract on rejection wouldn't be OK either. I'll wordsmith.

Reply:
Oh, I really meant "returned value" to be expansive: the promise and the value it resolves to OR the value it might produce on rejection.  But your suggestion works.